common/Util.cpp

```diff
@@ -125,16 +130,23 @@ namespace Util
 
             // a poor fallback but something.
             std::vector<char> random = getBytes(length);
-            int fd = open("/dev/urandom", O_RDONLY);
             int len = 0;
+#ifdef HAVE_SYS_RANDOM_H
+            len = getrandom(random.data(), length, GRND_NONBLOCK);
+            if(len < 0)
+            {
+                LOG_ERR("The requested entropy was not available: " << errno);
+            }
+#else
+            int fd = open("/dev/urandom", O_RDONLY);
```

Comment:
It would be great if we fall back to the "/dev/[u]random" approach, instead of failing altogether.